apps/communications/contacts/js/contacts.js

```diff
@@ -893,6 +888,9 @@ var Contacts = (function() {
     window.removeEventListener('DOMContentLoaded', onLoad);
   });
 
+  window.addEventListener('overlayshown', onOverlayShown);
+  window.addEventListener('overlayhidden', onOverlayHidden);
```

Comment:
What about if any other code within our app is calling `utils.overlay.show`? We will listen the set of events and execute the handler when we don't need it...  Could we do the same with a promise? Could we call this 'deletion' process from List.js, and wait for it with the Promise instead of events? Scope will be easier to read, and we are decoupling as well from contacts.js.